nikola/plugins/compile/markdown/__init__.py

```diff
@@ -134,3 +141,17 @@ def create_post(self, path, **kw):
                 fd.write(write_metadata(metadata))
                 fd.write('-->\n\n')
             fd.write(content)
+
+    def read_metadata(self, post, file_metadata_regexp=None, unslugify_titles=False, lang=None):
+        """Read the metadata from a post, and return a metadata dict."""
+        if not self.support_metadata:
+            return {}
+        if Markdown is None:
+            req_missing(['markdown'], 'build this site (compile Markdown)')
+        if lang is None:
+            lang = LocaleBorg().current_lang
+        # FIXME: what about translations????
+        source = post.source_path
```

Comment:
`post.translated_source_path(lang)`

Also, note that Pelican uses Title-Cased metadata fields, and has some different names.